### ISSUE TITLE
Cgl fails on OSX 10.11.5

### ISSUE THREAD

**Issue by enriquefernandez** [coin-or-tools/homebrew-coinor#38]
Formula fails complaining about Osi to being found. However, osi was already installed.

```
$ brew install cgl
==> Installing cgl from coin-or-tools/coinor
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Downloading http://www.coin-or.org/download/pkgsource/Cgl/Cgl-0.59.6.tgz
Already downloaded: /Users/efernan/Library/Caches/Homebrew/cgl-0.59.6.tgz
==> ./configure --prefix=/usr/local/Cellar/cgl/0.59.6 --datadir=/usr/local/Cellar/cgl/0.59.6/share/cgl --includedir=/usr/local/Cellar
Last 15 lines from /Users/efernan/Library/Logs/Homebrew/cgl/01.configure:
checking if /usr/local/bin/gfortran static flag -static works... rm: conftest.dSYM: is a directory
no
checking if /usr/local/bin/gfortran supports -c -o file.o... rm: conftest.dSYM: is a directory
yes
checking whether the /usr/local/bin/gfortran linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.5.0 dyld
checking how to hardcode library paths into programs... immediate
configure: Build is "i386-apple-darwin15.5.0".
configure: Applying patches to libtool for Darwin
checking if library version is set... 10:6:9
checking for pkg-config... pkg-config
checking pkg-config is at least version 0.16.0... yes
checking for COIN-OR package CoinUtils... yes: 2.10.10
checking for COIN-OR package Osi... not given: No package 'osi' found
configure: error: Required package Osi not available.

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/coin-or-tools/homebrew-coinor/issues
```

Osi was installed with homebrew:

```
$ brew install osi
Warning: coin-or-tools/coinor/osi-0.107.5 already installed
```

**Comment by tkralphs** [comment]
Seems like we need an explicit dependency on `osi`, since otherwise, pkg-config doesn't get the correct path and `configure` can't find `osi`. The theory was that since `clp` depends on `osi` and `cgl` depends on `clp`, the direct dependency on `osi` should be unnecessary, but this doesn't seem to work. This PR should fix it: #39.

**Comment by tkralphs** [comment]
Hmm, unfortunately, PR #39 only raised a new issue. See #40. Still working on it.

**Comment by dpo** [comment]
@tkralphs Sorry I haven't been active here recently. If CLP depends on OSI and CGL depends on CLP, why is CGL trying to detect OSI? Can it use OSI independently of CLP? If so, doesn't it have configure options to specify where to find OSI? (I'm not able to check right now.). I can experiment locally later.

**Comment by tkralphs** [comment]
@dpo Ah, OK, I just realized while trying to think of how to answer you questions that #40 is the real problem and this PR is probably just trying to fix the symptom of that. 

What determines whether a package is detected as installed at the moment is just what `brew` puts into the `PKG_CONFIG_PATH`, since COIN's `configure` uses `pkg-config` to detect other packages. Because `clp` doesn't detect the presence of `osi` properly (as described in #40), then `cgl` doesn't either. However, 'cgl` apparently needs `osi-clp` to build properly. 

Actually, as an aside, there is something wrong in the `cgl` configuration, since it doesn't seem to build without `osi-clp`, but the configuration succeeds even though `osi-clp` is not found. I will fix that. 

I think if we fix #40, then this PR will be unnecessary.

**Comment by tkralphs** [comment]
This is fixed now by #41.

**Comment by rickyars** [comment]
i have a problem where homebrew can't find pkg-config even though it's installed:
```
$ brew list pkg-config
/usr/local/Cellar/pkg-config/0.29.1_2/bin/pkg-config
/usr/local/Cellar/pkg-config/0.29.1_2/share/aclocal/pkg.m4
/usr/local/Cellar/pkg-config/0.29.1_2/share/doc/pkg-config/pkg-config-guide.html
/usr/local/Cellar/pkg-config/0.29.1_2/share/man/man1/pkg-config.1

$ pkg-config --version
0.29.1
```
but when i try to install symphony i get the following:
```
checking for pkg-config... no
checking for COIN-OR package CoinDepend... skipped check via pkg-config, redirect to fallback
checking for COIN-OR package CoinDepend (fallback)... no, dependency coinutils not available
configure: error: Required package CoinUtils or Osi not available.
```

i realize this might not be the right place for the question, but i don't know where else to ask
